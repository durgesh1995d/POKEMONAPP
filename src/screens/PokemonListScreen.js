import React, { useEffect, useRef, useState } from 'react';
import { Alert, Button, Dimensions, SafeAreaView, View } from 'react-native';
import PokemonList from '../component/PokemonList';
import * as Notifications from 'expo-notifications';

const { height, width } = Dimensions.get('window');

const PokemonListScreen = ({ navigation }) => {
  const requestPermissions = async () => {
    const { status } = await Notifications.getPermissionsAsync();
    if (status !== 'granted') {
      await Notifications.requestPermissionsAsync();
    }
  };

  useEffect(() => {
    requestPermissions();
  }, []);

  const scheduleNotification = async (minutes) => {
    const trigger = new Date(Date.now() + minutes * 60 * 1000); // Schedule for `minutes` later

    await Notifications.scheduleNotificationAsync({
      content: {
        title: 'Scheduled Notification',
        body: `This is a notification scheduled for ${minutes} minutes.`,
        sound: true,
        priority: Notifications.AndroidNotificationPriority.MAX, // Android: Ensure high priority
      },
      trigger,
    });
    Alert.alert(`Notification scheduled for ${minutes} minutes from now.`);
    console.log(`Notification scheduled for ${minutes} minutes from now.`);
  };
  Notifications.setNotificationHandler({
    handleNotification: async () => ({
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: true,
    }),
  });

  return (
    <SafeAreaView>
      <PokemonList navigation={navigation} />
      <View
        style={{
          position: 'absolute',
          bottom: 20,
          // alignItems: 'center',
          alignSelf: 'center',
          // justifyContent: 'center',
        }}
      >
        <View style={{ paddingBottom: 10 }}>
          <Button
            title='Schedule 2 Min Notification'
            onPress={() => scheduleNotification(2)}
          />
        </View>
        <Button
          title='Schedule 5 Min Notification'
          onPress={() => scheduleNotification(5)}
        />
      </View>
    </SafeAreaView>
  );
};

export default PokemonListScreen;
