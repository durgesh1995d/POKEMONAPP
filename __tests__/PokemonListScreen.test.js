// __tests__/PokemonListScreen.test.js
import React from 'react';
import { render } from '@testing-library/react-native';
import PokemonListScreen from '../src/screens/PokemonListScreen';

// Mock the API hook
jest.mock('../src/api/pokemonApi', () => ({
  useGetPokemonListQuery: jest.fn().mockReturnValue({}),
}));

// Mock the API hook
jest.mock('@reduxjs/toolkit/query/react', () => {
  return { useGetPokemonByNameQuery: jest.fn() };
});

test('renders PokemonListScreen correctly', () => {
  const navigation = { navigate: jest.fn() }; // Mock navigation prop

  const { getByTestId, toJSON } = render(
    <PokemonListScreen navigation={navigation} />
  );

  expect(getByTestId('PokemonList')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});
