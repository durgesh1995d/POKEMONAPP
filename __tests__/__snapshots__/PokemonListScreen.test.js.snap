// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokemonListScreen correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    testID="PokemonList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
